src/sdi-notify.c

```diff
@@ -0,0 +1,278 @@
+  }
+
+  show_pending_update_notification(self, title, body, icon,
+                                   g_slist_append(NULL, snap));
+}
+
+void sdi_notify_pending_refresh_multiple(SdiNotify *self, GSList *snaps) {
+  g_return_if_fail(SDI_IS_NOTIFY(self));
+  g_return_if_fail(snaps != NULL);
+
+  g_autoptr(GString) body = g_string_new(_("Close the apps to start updating"));
+  for (; snaps != NULL; snaps = snaps->next) {
+    SnapdSnap *snap = (SnapdSnap *)snaps->data;
+    const gchar *name = snapd_snap_get_name(snap);
+    g_autoptr(GAppInfo) app_info = sdi_get_desktop_file_from_snap(snap);
+    if (app_info != NULL) {
+      name = g_app_info_get_display_name(app_info);
+    }
+
+    GTimeSpan difference = get_remaining_time(snap) / 1000000;
+    if (difference > 86400) {
+      g_string_append_printf(body, _(" (%s %ld days left)"), name,
+                             difference / 86400);
+    } else if (difference > 3600) {
+      g_string_append_printf(body, _(" (%s %ld hours left)"), name,
+                             difference / 3600);
+    } else {
+      g_string_append_printf(body, _(" (%s %ld minutes left)"), name,
+                             difference / 60);
+    }
+  }
+  g_autoptr(GIcon) icon = g_themed_icon_new("emblem-important-symbolic");
+  show_pending_update_notification(self, _("Pending updates for some snaps"),
+                                   body->str, g_steal_pointer(&icon), snaps);
+}
+
+void sdi_notify_refresh_complete(SdiNotify *self, SnapdSnap *snap,
+                                 const gchar *snap_name) {
+  g_return_if_fail(SDI_IS_NOTIFY(self));
+  g_return_if_fail((snap != NULL) || (snap_name != NULL));
+
+  GIcon *icon = NULL;
+  g_autoptr(GAppInfo) app_info = NULL;
+  const gchar *name = NULL;
+
+  if (snap != NULL) {
+    app_info = sdi_get_desktop_file_from_snap(snap);
+    if (app_info != NULL) {
+      name = g_app_info_get_display_name(app_info);
+      icon = g_app_info_get_icon(app_info);
+    }
+    if (name == NULL)
+      name = snapd_snap_get_name(snap);
+  }
+  if (name == NULL)
+    name = snap_name;
+
+  g_autofree gchar *title = g_strdup_printf(_("%s was updated"), name);
+
+  show_simple_notification(self, title, _("Ready to launch"), icon,
+                           "update-complete");
+}
+
+GApplication *sdi_notify_get_application(SdiNotify *self) {
+  g_return_val_if_fail(SDI_IS_NOTIFY(self), NULL);
+  return self->application;
+}
+
+static void sdi_notify_action_ignore(GActionGroup *action_group,
+                                     GVariant *app_list, SdiNotify *self) {
+  gsize len;
+  g_autofree gchar **apps = (gchar **)g_variant_get_strv(app_list, &len);
+
+  g_return_if_fail(apps != NULL);
+  for (gsize i = 0; i < len; i++)
+    g_signal_emit_by_name(self, "ignore-snap-event", apps[i]);
+}
+
+static void sdi_notify_action_close(GActionGroup *action_group,
+                                    GVariant *action_name, SdiNotify *self) {
+  g_print("Action2 close\n");
```

Comment:
Left over debugging? Perhaps change to a g_warning/g_debug if you want this to remain.

Reply:
Yes, a left over O:-) Removing.